src/api/java/com/minecolonies/api/entity/citizen/AbstractEntityCitizen.java

```diff
@@ -362,6 +365,21 @@ public void push(@NotNull final Entity entityIn)
         super.push(entityIn);
     }
 
+    @Override
+    public void onPlayerCollide(final Player player)
+    {
+        final IJob job = getCitizenData().getJob();
+        if (job == null || job.getJobRegistryEntry() != ModJobs.archer && job.getJobRegistryEntry() != ModJobs.knight && job.getJobRegistryEntry() != ModJobs.druid)
```

Comment:
Perhaps a good time for `job.isGuard()` or sth?  Elsewhere an instanceof Guard check is used.

Reply:
ye if we need this again I'll generalise it

Reply:
Again though, it's already checked as `job instanceof AbstractJobGuard` elsewhere, so this ought to be consistent.  And that's more future-proof/addon-proof for new guard types.

Reply:
abstractjobguard is not in the api

Reply:
Though IJob could have an isGuard method, which is overridden in AbstractJobGuard to return true instead of false. I don't think that would cause problems with that class not being in the API